packages/core/src/App/Containers/RealAccountSignup/set-currency.jsx

```diff
@@ -0,0 +1,112 @@
+import classNames from 'classnames';
+import React from 'react';
+import { localize } from '@deriv/translations';
+import { website_name } from 'App/Constants/app-config';
+import { connect } from 'Stores/connect';
+import CurrencySelector from './currency-selector.jsx';
+import { generateValidationFunction } from './form-validations';
+import { currency_selector_config } from './currency-selector-form';
+import 'Sass/set-currency.scss';
+import 'Sass/change-account.scss';
+
+class SetCurrency extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            form_error: '',
+            form_value: {
+                currency: '',
+            },
+        };
+    }
+
+    clearError = () => this.setState({ form_error: '' });
+
+    setCurrency = (obj, setSubmitting) => {
+        this.props.onLoading();
+        const { currency } = obj;
+        if (currency) {
+            this.props
+                .setCurrency(currency)
+                .then(response => {
+                    setSubmitting(false);
+                    this.props.onSuccessSetAccountCurrency('', response.echo_req.set_account_currency);
+                })
+                .catch(error_message => {
+                    this.props.onError(error_message);
+                });
+        }
+    };
+
+    updateValue = (index, value, setSubmitting) => {
+        this.setCurrency(value, setSubmitting);
+    };
+
+    get no_crypto_available() {
+        return this.props.available_crypto_currencies.length === 0 && this.props.has_fiat;
+    }
+
+    get should_hide_crypto() {
+        return this.props.is_eu_enabled && this.props.is_eu; // TODO [deriv-eu] remove is_eu_enabled once released
+    }
+
+    render() {
+        return (
+            <div
+                className={classNames('set-currency-modal', {
+                    'set-currency-modal--disabled':
+                        this.props.available_crypto_currencies.length === 0 && this.props.has_fiat,
```

Comment:
isn't this condition logic already added in the `no_crypto_available()` getter?

Reply:
Yes. We can reuse it here.